h5py/__init__.py

```diff
@@ -38,11 +38,12 @@
             '{0}.{1}.{2}'.format(*version.hdf5_built_version_tuple)
     ))
 
-
 _errors.silence_errors()
 
-from ._conv import register_converters as _register_converters
+from ._conv import register_converters as _register_converters, \
+                   unregister_converters as  _unregister_converters
 _register_converters()
+atexit.register(_unregister_converters)
```

Comment:
Is this related to the rest of the PR?

Reply:
In theory not. In practice Python seg-faults at output without this patch.
Actually h5py was not unregistering the conversion filters from hdf5 at exit. Everything was there except it was not done.
We found out that once this "nogil" feature is activated, the unregistration is called after the gil itself is destroyed, hence the seg-fault.